packages/mdc-theme/_variables.scss

```diff
@@ -23,11 +23,13 @@
 //
 
 $mdc-theme-primary: #6200ee !default; // baseline purple, 500 tone
+$mdc-theme-on-primary: #fff !default;
```

Comment:
Shouldn't these still be computed based on `$mdc-theme-primary` or `$mdc-theme-secondary`?

i.e. these values should adjust to remain correct when the corresponding variable is changed, which we were doing before and should still easily be able to do by calling a mixin here.

Also, are we really doing away with semitransparent text OOTB? I thought that was in spec. I thought on-primary/secondary were additional slots, not replacing text?

Reply:
mdc-theme should not compute `$mdc-theme-on-primary` based on `$mdc-theme-primary` because then `$mdc-theme-on-primary` is always white/black. Mdc-theme should allow users to customize `$mdc-theme-on-primary` to any color, like the brown color used in the shrine example.

Reply:
Sorry if I worded poorly; my suggestion wouldn't prevent users from customizing it, it would just provide a sane default.

We'd still have `!default` at the end to allow overriding it.

Reply:
I cant actually figure how to build this functionality (without probably restructuring all our mixins in mdc-theme). I tried to do this

```
$mdc-theme-on-primary: #fff !default;
@if mdc-theme-contrast-tone($mdc-theme-primary) == "dark" {
  $mdc-theme-on-primary: #000 !default;
}
```

and commented out the `$mdc-theme-on-primary` in `theme-shrine.scss`...but it keeps setting the -on-primary color the white

Reply:
That will fail because the earlier assignment will always happen, and then the latter one inside the if will never happen because `!default` makes the assignment only happen if that variable doesn't already have an assigned value. This is the exact behavior that makes redefining our globals before loading our styles work.

http://sass-lang.com/documentation/file.SASS_REFERENCE.html#variable_defaults_default

There is an `if` function that can be used akin to a ternary operator in other languages which you can use to set the value in one statement:

```scss
$mdc-theme-on-primary: if(mdc-theme-contrast-tone($mdc-theme-primary) == "dark", #000, #fff) !default;
```

Or, you can populate a map with `light` and `dark` entries and simply reference that with the value returned from contrast-tone. That is what the logic that you're replacing is _already doing_, which makes it hard for me to understand why we're removing it.

https://github.com/material-components/material-components-web/blob/v0.33.0/packages/mdc-theme/_variables.scss#L42-L63

Reply:
How would users override the -on-primary with Sass without the changes in this PR?

Reply:
IIUC your point is right now it's impossible to directly override, say, text-primary-on-primary, which is true. If enabling that is the goal, then I can see why this PR is changing how the variables are defined, so they can each be _directly_ overridden.

I was specifically concerned about the existing dark/light logic that was being taken out of the picture in the process of this change, and was a little confused since in this thread you seemed to be having trouble replicating the logic that already exists.

How many text color levels are we going to end up with after this is all done (compared to the current primary/secondary/hint/disabled/icon)? Would it be cleaner to reuse the existing functions that pick the defaults out of the `mdc-theme-text-colors` map, versus writing ternaries with hard-coded values on every variable we declare?

Either way, we should be able to ensure sane behavior if someone doesn't touch the on-x variables, but _does_ touch the primary/secondary values.